pkg/clean/clean.go

```diff
@@ -27,19 +27,25 @@ var log = logger.Log
 // including security policy, static route, subnet, subnet port, subnet set, vpc, ip pool, nsx service account
 // it is usually used when nsx-operator is uninstalled and remove all the resources created by nsx-operator
 // return error if any, return nil if no error
-func Clean(cf *config.NSXOperatorConfig) (Status, error) {
+// the error type include followings:
+//
+//	 	ValidationFailed 			indicate that the config is incorrect and failed to pass validation
+//		GetNSXClientFailed  		indicate that could not retrieve nsx client to perform cleanup operation
+//		InitCleanupServiceFailed 	indicate that error happened when trying to initialize cleanup service
+//		CleanupResourceFailed    	indicate that the cleanup operation failed at some services, the detailed will in the service logs
+func Clean(cf *config.NSXOperatorConfig) error {
 	log.Info("starting NSX cleanup")
 	if err := cf.ValidateConfigFromCmd(); err != nil {
-		return ValidationFailed, err
+		return ValidationFailed
 	}
 	nsxClient := nsx.GetClient(cf)
 	if nsxClient == nil {
-		return GetNSXClientFailed, fmt.Errorf("failed to get nsx client")
+		return GetNSXClientFailed
 	}
 	if cleanupService, err := InitializeCleanupService(cf); err != nil {
-		return InitCleanupServiceFailed, fmt.Errorf("failed to initialize cleanup service: %w", err)
+		return InitCleanupServiceFailed
 	} else if cleanupService.err != nil {
```

Comment:
Not related to your change, but - why does `cleanupService` have a property called `err`? This is pretty uncommon to see.

Reply:
@ihgann refer https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/clean/clean.go#L82, the purpose is to avoid error check hell.